src/panel/pages/timeline/Timeline.tsx

```diff
@@ -1,39 +1,58 @@
-import React, { FC } from "react";
+import React, { FC, useMemo } from "react";
 import styled from "styled-components";
 import { useTimelineContext } from "../../context";
 import { Background } from "../../components/Background";
-import { TimelineRow, TimelinePane } from "./components";
+import { TimelineRow, TimelinePane, Tick } from "./components";
 
 export const Timeline: FC = () => {
   const {
     setContainer,
-    getTimePosition,
+    scale,
     events,
+    startTime,
+    container,
     selectedEvent
   } = useTimelineContext();
 
+  const ticks = useMemo(
+    () =>
+      scale &&
+      scale.ticks(getTickCount(container.clientWidth)).map(t => ({
+        label: `${t - startTime}ms`,
+        position: scale(t)
+      })),
+    [scale]
+  );
+
   // We lie about the types to save having to do this check
   // in every component. This guard is needed.
-  if (!getTimePosition)
+  if (!container)
     return (
-      <Container>
+      <Page>
         <TimelineList ref={setContainer} />
-      </Container>
+      </Page>
     );
 
   return (
-    <Container>
+    <Page>
       <TimelineList ref={setContainer}>
+        {ticks.map(t => (
+          <Tick
```

Comment:
From the useMemo above it looks like ticks could be undefined - do we need to a truthy check here or will it be okay?

Reply:
Good shout - fixed!